test/one_hot_encoding.jl

```diff
@@ -174,14 +174,14 @@
 
         df = DataFrame(:a => ["foo", "bar"], :b => ["foo2", "bar2"])
         expected = DataFrame(
-            [[1, 0], [0, 1], [0, 0], [0, 0], [0, 0], [0, 0], [0, 0], [0, 0], [1, 0], [0, 1]],
-            [Symbol.(:Column, x) for x in 1:10],
+            Bool[1 0 0 0 0; 0 1 0 0 0; 0 0 0 1 0; 0 0 0 0 1],
+            [Symbol.(:Column, x) for x in 1:5],
```

Comment:
interestingly this use-case changed when I made the refactoring. I'm not too concerned about this because the previous result was inconsistent with the matrix application. This version is now consistent.

Secondly, I'm not sure how much sense to apply OHE  to multiple columns. The output isn't necessarily valid or useful, so I would considering dropping the tests on multi-dimensional array altogether.

Reply:
Could you explain more what was going on before and how it works now? I'm just pressed for time.

Reply:
I think what was happening before is (in the Tables `apply`) when we had multiple columns we were calling `_apply` on each component in turn and then stacking the result. This lead to the original version that had many columns, which would have had duplicate names.

Now instead, we collect the components and call `_apply` on the collection, which is what we were doing for the `Array` method all along. So now we get one column per category (as expected) but the shape is now inconsistent.

This is what I mean by it doesn't make sense to call OHE on a Matrix. I think in a future release I'm going to delete these tests because they don't seem worth supporting.